quic/s2n-quic/src/lib.rs

```diff
@@ -50,6 +50,18 @@
 //!
 //! **NOTE**: this will override the platform detection and always use [`s2n-tls`][s2n-tls] by default.
 //!
+//! ## Unstable features
+//!
+//! These features enable **unstable** features. Unstable features are subject to change without
+//! notice. To enable these features, the `--cfg s2n_quic_unstable` option must be passed to
+//! rustc when compiling. This is easiest done using the RUSTFLAGS env variable:
+//! `RUSTFLAGS=\"--cfg s2n_quic_unstable\"`.
+//!
+//! ### `unstable_client_hello`
+//!
+//! Enables the `ClientHelloHandler` trait, which can be used to set the client_hello callback on
+//! s2n-tls provider.
+//!
```

Comment:
Trying to decide if we want to document unstable features at this point

Reply:
yea i was leaning towards no and not advertise it. if there is a usecase/question about it then we can always add some documentation around it. currently we do have ` // disable warning for CI` comment